### PR TITLE
SQL PATH for CLI/JDBC/Client

### PR DESCRIPTION
**Pull request by wendigo**
Adds support for https://trino.io/docs/current/sql/set-path.html

+ fixes for timeout

Proposed release entry:

```
Add support for setting default SQL PATH in the CLI and JDBC driver
```

### CODE REVIEW

File: client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java [hunk 1]
```diff
@@ -210,6 +211,10 @@ public class ClientOptions
     @Option(names = "--schema", paramLabel = "<schema>", description = "Default schema")
     public Optional<String> schema;
 
+    @PropertyMapping(SQL_PATH)
+    @Option(names = "--path", paramLabel = "<schema>", description = "Default SQL path")
```
Comment (nineinchnick) [inline client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:215]: `SET PATH` SQL query supports passing multiple values, should we support it here too?

**Review by losipiuk — APPROVED** [review]

File: client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java (hunk 1 above)
Comment (losipiuk) [reply client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:215]: And then it is searched in order?

Comment (losipiuk) [reply client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:215]: Indeed - nice catch

Comment (wendigo) [reply client/trino-cli/src/main/java/io/trino/cli/ClientOptions.java:215]: I've fixed the API in different places.

**Review by nineinchnick — APPROVED** [review]

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java [hunk 2]
```diff
@@ -280,6 +281,20 @@ public void testUriWithSslEnabled()
...
 
+    @Test
+    public void testSqlPath()
+            throws SQLException
+    {
+        assertInvalid("jdbc:trino://localhost:8080?path=catalog.schema.whatever", "Connection property 'path' has invalid syntax, should be [catalog].[schema] or [schema]");
+        assertInvalid("jdbc:trino://localhost:8080", properties("path", "catalog.schema.whatever"), "Connection property 'path' has invalid syntax, should be [catalog].[schema] or [schema]");
+
+        assertThat(createDriverUri("jdbc:trino://localhost:8080?path=catalog.schema").getPath()).hasValue(ImmutableList.of("catalog.schema"));
+        assertThat(createDriverUri("jdbc:trino://localhost:8080?path=catalog,catalog2").getPath()).hasValue(ImmutableList.of("catalog", "catalog2"));
```
Comment (martint) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java:292]: SQL path is a list of schemas, so the names used in these tests are misleading. Change to “schema1”, “schema2”, etc?

Comment (wendigo) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java:292]: it's catalog and schema or just schema (at least according to the docs):
```
Define a collection of paths to functions or table functions in specific catalogs and schemas for the current session.

Each path-element uses a period-separated syntax to specify the catalog name and schema location <catalog>.<schema> of the function, or only the schema location <schema> in the current catalog. The current catalog is set with [USE](https://trino.io/docs/current/sql/use.html), or as part of a client tool connection. Catalog and schema must exist.
```

Comment (martint) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java:292]: Yes, that's what I meant by "schema" -- the leaf elements have to be schemas. You can't have a naked catalog in the path.

Comment (wendigo) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverUri.java:292]: Makes sense :)

**Pull request merged**